.gitlab-ci.yml

```diff
@@ -192,14 +194,56 @@ build-windows:
 
 publish-docker:
   stage:                           publish
-  only:                            *releaseable_branches
-  cache:                           {}
   dependencies:
     - build-linux
   tags:
-    - shell
+    - kubernetes-parity-build
+  environment:
+    name: parity-build
+  cache:                           {}
+  only:                            *releaseable_branches
+  except:
+    - master
+  image:                           docker:stable
+  services:
+    - docker:dind
+  variables:
+    DOCKER_HOST:                   tcp://localhost:2375
+    DOCKER_DRIVER:                 overlay2
+    GIT_STRATEGY:                  none
+    # DOCKERFILE:                  artifacts/Dockerfile
+    CONTAINER_IMAGE:               parity/parity
+  before_script:
+    - test "$Docker_Hub_User_Parity" -a "$Docker_Hub_Pass_Parity"
+        || ( echo "no docker credentials provided"; exit 1 )
+    - docker login -u "$Docker_Hub_User_Parity" -p "$Docker_Hub_Pass_Parity"
+    - docker info
   script:
-    - scripts/gitlab/publish-docker.sh parity
+    # we stopped pushing nightlies to dockerhub, will push to own registry prb.
+    - case "${SCHEDULE_TAG:-${CI_COMMIT_REF_NAME}}" in
+        $SCHEDULE_TAG)
+          echo "Docker TAG - ${CONTAINER_IMAGE}:${SCHEDULE_TAG}";;
```

Comment:
so no docker build for schedules with $SCHEDULE_TAG like nightly?

Reply:
Yes, that's the idea. Most probably they are not needed to anyone.
Maybe we'll reintroduce nightly builds later, when our own repository will start working.